alex's tasks

```diff
@@ -0,0 +1,59 @@
+1 Run plain ruby code
+with irb
+with ruby command + .rb file
+use require and require_relative to load code from another file
+
+2 Know where the official ruby documentation is and actually use it
+ruby-doc.org
+
+3 Built-in types
+Strings
+Symbols
+Arrays and Hashes
+Add elements
+Remove elements
+Access elements
+Iterate through elements
+Handful of methods from Enumerable
+#all?
+#any?
+#collect` / `#map`
+#each_with_index and #each_with_object
+#inject / #reduce
+#select and #reject
+#uniq
+implement <=> operator for #min #max #sort
+Able to determine whether an object is able to respond to a specific method
+
+4 Simple methods (actually functions: not within a class)
+Able to define your own methods
+arguments: standard way, optional arguments, keyword arguments and variable arguments
+return value
+
+5 Control Flow
+if/else
+case statement
+loops: each, while
+raise and handle errors
+add control-flow to your methods
+
+6 Objects and Methods
+Understand the difference between instances and classes
+Define your own classes
+Able to write and apply instance methods effectively
+Able to write and apply class methods effectively
+Able to write constructors (def initialize)
+with and without arguments
+Inheritance
+Inherit methods from parents
+Overwrite parents' methods
+
+7 scope and visibility
+local, instance and class variables (forget about global)
+self
```

Comment:
class Point 
attr_reader :x :y

def **self**.sum(*points) 
x = y = 0 
end 

`self` uses the DRY (don't repeat yourself) principle.